app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java

```diff
@@ -259,6 +267,29 @@ private void setFontColor() {
         materialDialog.show();
     }
 
+    private void setPageColor() {
```

Comment:
`setPageColor()` method is repeated once. It should be reused and probably be placed in `CommonCodeUtils` (with a javadoc style comment of course).

Reply:
both of them look similar but are not. they set different values in shared pref. and on click of Ok, they also set the value of mPageColor. so if we move them to different util class then we will have to somehow return the current color selected to the appropriate fragment

Reply:
@Ni3verma Have you tried extracting the dialog's common code?

```
 public static void showPageColorDialog(Activity activity, MaterialDialog.SingleButtonCallback callback){
        MaterialDialog materialDialog = new MaterialDialog.Builder(activity)
                .title(R.string.page_color)
                .customView(R.layout.dialog_color_chooser, true)
                .positiveText(R.string.ok)
                .negativeText(R.string.cancel)
                .onPositive(callback)
                .build();

        ColorPickerView colorPickerView = materialDialog.getCustomView().findViewById(R.id.color_picker);
        colorPickerView.setColor(colorPickerView.getColor());
        materialDialog.show();
    }
```

Reply:
even If we extract the common code, what will we do after that?
I guess we will have to add new listener interfaces and through callbacks we will have to carry back selected color to the fragment.
so I thought it's better to do it this simple way